x-pack/plugins/apm/public/components/shared/links/apm/transaction_detail_link/index.tsx

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiFlexGroup, EuiFlexItem, EuiLink, EuiText } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { identity, pickBy } from 'lodash';
+import React from 'react';
+import { useLocation } from 'react-router-dom';
+import { pickKeys } from '../../../../../../common/utils/pick_keys';
+import { useApmPluginContext } from '../../../../../context/apm_plugin/use_apm_plugin_context';
+import { useLegacyUrlParams } from '../../../../../context/url_params_context/use_url_params';
+import { unit } from '../../../../../utils/style';
+import { PopoverTooltip } from '../../../popover_tooltip';
+import { getComparisonEnabled } from '../../../time_comparison/get_comparison_enabled';
+import { TruncateWithTooltip } from '../../../truncate_with_tooltip';
+import { APMQueryParams } from '../../url_helpers';
+import { APMLinkExtendProps, getLegacyApmHref } from '../apm_link';
+
+export const txGroupsDroppedBucketName = '_other';
+
+interface Props extends APMLinkExtendProps {
+  serviceName: string;
+  traceId?: string;
+  transactionId?: string;
+  transactionName: string;
+  transactionType: string;
+  latencyAggregationType?: string;
+  environment?: string;
+  comparisonEnabled?: boolean;
+  offset?: string;
+  overflowCount?: number;
+}
+
+const persistedFilters: Array<keyof APMQueryParams> = [
+  'transactionResult',
+  'serviceVersion',
+];
+
+export function TransactionDetailLink({
+  serviceName,
+  traceId,
+  transactionId,
+  transactionName,
+  transactionType,
+  latencyAggregationType,
+  environment,
+  comparisonEnabled,
+  offset = '1d',
+  overflowCount = 0,
+  ...rest
+}: Props) {
+  const { urlParams } = useLegacyUrlParams();
```

Comment:
Can this be changed to use the new hook?

Reply:
Good point. We should aim to use `useApmParams`. See https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/dev_docs/routing_and_linking.md#client-side